models/animals.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+
+class Animal(ABC):
+    def __init__(self, name: str, weight: float):
+        self._name = name
+        self._weight = weight
+
+    def __str__(self) -> str:
+        return f"My name: {self._name}. My weight: {self._weight}. "
+
+    @abstractmethod
+    def who_are_u(self) -> str:
```

Comment:
Alternative way to do this without overriding this method for all the classes: https://stackoverflow.com/questions/510972/getting-the-class-name-of-an-instance
But not a problem though, the way you did it is completely fine.
It's just an idea for your curiosity. Because when making a base class (the one that inherits from abc.ABC) in Python, you're _**not really**_ obliged to add abstract methods there.